src/middleware.ts

```diff
@@ -362,21 +371,15 @@ export const persist =
         }
       })
       .then(() => {
-        postRehydrationCallback?.(stateFromStorageInSync, undefined)
+        postRehydrationCallback?.(
+          { ...configResult, ...stateFromStorageInSync },
```

Comment:
Yeah, this is something I concerned with "merged" state. Looks good.
One caveat is L375 and L384 are different objects. Would be nice to make it referentially identical.

Reply:
good idea

Reply:
there you go :slightly_smiling_face: